src/app/submission/pages/submission.component.ts

```diff
@@ -277,18 +277,21 @@ export class SubmissionComponent implements OnInit, OnDestroy {
     this.downloadDisabled = true;
     const uuid = this.submissionEnvelope['uuid']['uuid'];
     this.loaderService.display(true, 'This may take a moment. Please wait...');
+    // now set a cookie to disable further download attempt within X min of this request or
+    // until a response is returned when the cookie gets deleted.
+    this.cookieService.set(this.DOWNLOAD_TIMEOUT_COOKIE, "1", {expires: this.DOWNLOAD_BACKOFF_MINS / (24*60)})
     this.brokerService.downloadSpreadsheet(uuid).subscribe(response => {
       const filename = response['filename'];
       const newBlob = new Blob([response['data']]);
         this.saveFile(newBlob, filename);
         this.loaderService.display(false);
       this.downloadDisabled = false;
+      this.cookieService.delete(this.DOWNLOAD_TIMEOUT_COOKIE);
```

Comment:
This line broke the unit tests but I've [fixed them](https://github.com/ebi-ait/ingest-ui/commit/a26bbadacc2f1a6a73605e01d8b9672559cf7523).